### DISCUSSION TITLE
Global package.json

### DISCUSSION THREAD

**Discussion by Aprillion** [OpenSourceRaidGuild/website#12] [General]
We have package.json, README, lock file, LICENSE and ~10 config files inside `frontend` directory.

I would prefer to have only 1 set of all config files in the repo, shared for frontend and any future directories by default - until there is a real need to separate anything.

Benefits of shared config:
* `yarn` on root level to install everything
* no mis-aligned config between directories when someone hacks together config for different tools
* single point of (updating) documentation and license

**Comment by nobrayner** [comment]
Thanks for bringing this up! I know I would have completely missed it until it was too late 😅

Are there any potential downsides to doing this that we should know up front? I'm assuming you have dealt with this before...?

**Reply by Aprillion** [reply]
I don't see anything that could not be split later if we decide we cannot live with common config, but need some 2 different configs for some tool ¯\\\_(ツ)_/¯

**Comment by sudokoi** [comment]
Will we be using `yarn workspaces`  or/and `lerna` for managing the monorepo?

Also, ide configs like `.vscode` should be in the parent directory.

**Reply by Aprillion** [reply]
I would advice to keep it as a simple single git repo, no monorepo tools until we experience *actual* problems that can be solved by extra monorepo solutions.
